pallets/schemas/src/lib.rs

```diff
@@ -625,19 +1002,57 @@ pub mod pallet {
 		/// Retrieve a schema info by id
 		pub fn get_schema_info_by_id(schema_id: SchemaId) -> Option<SchemaInfoResponse> {
 			if let Some(schema_info) = SchemaInfos::<T>::get(schema_id) {
-				let saved_settings = schema_info.settings;
-				let settings = saved_settings.0.iter().collect::<Vec<SchemaSetting>>();
 				let response = SchemaInfoResponse {
 					schema_id,
+					intent_id: schema_info.intent_id,
 					model_type: schema_info.model_type,
 					payload_location: schema_info.payload_location,
-					settings,
+					settings: schema_info.settings.0.iter().collect::<Vec<IntentSetting>>(),
+					status: schema_info.status,
 				};
 				return Some(response);
 			}
 			None
 		}
 
+		/// Retrieve an Intent by its ID
+		pub fn get_intent_by_id(
+			intent_id: IntentId,
+			include_schemas: bool,
+		) -> Option<IntentResponse> {
+			if let Some(intent_info) = IntentInfos::<T>::get(intent_id) {
+				let saved_settings = intent_info.settings;
+				let settings = saved_settings.0.iter().collect::<Vec<IntentSetting>>();
+				let schema_ids = include_schemas.then_some(
+					SchemaInfos::<T>::iter()
```

Comment:
This is dangerous if used incorrectly. I understand that include_schemas is false for all on-chain calls but we might need to add some kind of feature flag or at least a comment that should never be called with `include_schemas = true` from chain side.